projects/driver_controls/src/digital_io.c

```diff
@@ -29,100 +30,89 @@ typedef struct DigitalIOData {
 typedef struct DigitalIOSettings {
   GPIOAddress address;
   InterruptEdge edge;
-  DigitalIOData *data;
 } DigitalIOSettings;
 
 // Index the objects using their respective pins
 static DigitalIOData s_input_data[DRIVER_IO_NUM_ADDRESSES];
 
 // Genarate the event based on the identity of the triggering device
-static uint16_t prv_get_event(DigitalIOData *digital_io_data, GPIOState state) {
-  uint16_t event_id = digital_io_data->event;
+static uint16_t prv_get_event(DigitalIOData *data, GPIOState state) {
+  uint16_t event_id = data->event;
 
-  if (event_id == DIGITAL_IO_DEVICE_DIRECTION_SELECTOR && state == GPIO_STATE_HIGH) {
+  if (data->id == DIGITAL_IO_DEVICE_DIRECTION_SELECTOR && state == GPIO_STATE_HIGH) {
     event_id = INPUT_EVENT_DIRECTION_SELECTOR_NEUTRAL;
-  } else if (event_id == DIGITAL_IO_DEVICE_TURN_SIGNAL && state == GPIO_STATE_HIGH) {
+  } else if (data->id == DIGITAL_IO_DEVICE_TURN_SIGNAL && state == GPIO_STATE_LOW) {
     event_id = INPUT_EVENT_TURN_SIGNAL_NONE;
   }
 
   return event_id;
 }
 
 static void prv_input_callback(const GPIOAddress *address, void *context) {
-  DigitalIOData *data = context;
+  DigitalIOData *data = (DigitalIOData *)context;
 
   GPIOState state = { 0 };
   gpio_get_value(address, &state);
-  event_raise(prv_get_event(data, state), 0);
-}
-
-static void prv_init_pin(DigitalIOSettings *settings, GPIOSettings *gpio_settings) {
-  InterruptSettings it_settings = {
-    INTERRUPT_TYPE_INTERRUPT,  //
-    INTERRUPT_PRIORITY_LOW,    //
-  };
 
-  gpio_init_pin(&settings->address, gpio_settings);
-  gpio_it_register_interrupt(&settings->address, &it_settings, settings->edge, prv_input_callback,
-                             settings->data);
+  event_raise(prv_get_event(data, state), 0);
```

Comment:
Probably want to explicitly document what the data value vs event ID is supposed to represent. (Also, is the separate function really necessary at this point?)